troposphere/static/js/components/common/SSHCreateView.jsx

```diff
@@ -0,0 +1,182 @@
+import React from "react";
+
+export default React.createClass({
+    displayName: "SSHCreateView",
+
+    getInitialState: function() {
+        return {
+            keyName: "",
+            publicKey: "",
+            showValidation: false
+        };
+    },
+
+    validateName: function () {
+        let name = this.state.keyName;
+        let hasError = false;
+        let message = "";
+        let maxCharLen = 60;
+
+        if (name === "") {
+            hasError = true;
+            message = "This field is required";
+        }
+
+        if (name.length > maxCharLen) {
+            hasError = true;
+            message = `Must be less than ${maxCharLen} characters long`;
+        }
+
+        return {
+            hasError,
+            message
+        }
+    },
+
+    validatePublicKey: function () {
+        let pubKey = this.state.publicKey;
+        let parts = pubKey.split(/\s+/g)
+        let hasError = false;
+        let message = "";
+
+
+        if (pubKey === "") {
+            hasError = true;
+            message = "This field is required";
+        }
+
+        if (!this.validateKeyType(parts[0])) {
+            hasError = true;
+            message = "Public key must begin with either ssh-rsa, ssh-dss, ecdsa-sha2-nistp256, or ssh-ed25519";
+        }
+
+        return {
+            hasError,
+            message
+        }
+    },
+
+    validateKeyType: function(firstWord) {
+        return /^(ssh-dss|ecdsa-sha2-nistp256|ssh-ed25519|ssh-rsa)$/.test(firstWord);
+    },
+
+    validateKey: function() {
```

Comment:
Hmm, looks to me like `parts` used in line 48 is created in line 38 inside that same `validatePublicKey` function.

Maybe I'm not reading the javascript correctly (very possible) but it looks to me like this function `validateKey` is defined here but never used.  @robbaronbu?